### PR TITLE
Prefix protected fields with underscore for the internal XmlRawWriter

### PR DESCRIPTION
**Pull request by Marusyk**
Add underscore prefix to protected fields for the internal `XmlRawWriter` and its descendant classes 

Issue #35320

Please review
Thank you in advance

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawTextWriterGenerator.ttinclude [hunk 1]
```diff
@@ -250,10 +250,10 @@ namespace System.Xml
                 RawText("1.0");
 
                 // Encoding
-                if (encoding != null)
+                if (_encoding != null)
                 {
-                    RawText("\" encoding=\"");
-                    RawText(encoding.WebName);
+                    RawText("\" _encoding=\"");
```
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawTextWriterGenerator.ttinclude:255]: This is string output, believe shouldn't have `_`

File: src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawTextWriterGenerator.ttinclude [hunk 2]
```diff
@@ -147,66 +147,66 @@ namespace System.Xml
...
 
             // the buffer is allocated will OVERFLOW in order to reduce checks when writing out constant size markup
             if (settings.Async)
             {
-                bufLen = ASYNCBUFSIZE;
+                _bufLen = ASYNCBUFSIZE;
             }
 
-            <#= BufferName #> = new <#= BufferType #>[bufLen + OVERFLOW];
+            <#= BufferName #> = new <#= BufferType #>[_bufLen + OVERFLOW];
```
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawTextWriterGenerator.ttinclude:159]: seems `<#= BufferName #>` need to be updated here https://github.com/dotnet/runtime/blob/b0721f1f8415bc5064f898a1e94747ca8965f009/src/libraries/System.Private.Xml/src/System/Xml/Core/RawTextWriterEncoded.ttinclude#L8

Comment (buyaa-n) [reply src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawTextWriterGenerator.ttinclude:159]: might be worth run all template for all generated files and build

**Review by buyaa-n — CHANGES_REQUESTED** [review]
@Marusyk thank you for working on this, added couple of comments for template files, overall looks good to me

**Review by buyaa-n — APPROVED** [review]
Thank you @Marusyk LGTM, please merge conflicts

**Comment by krwq** [comment]
(restarting CI)

**Pull request merged**